golang/pkg/rnr/tasks_test.go

```diff
@@ -64,7 +72,33 @@ func (m *mockTask) Proto(updater func(*pb.Task)) *pb.Task {
 func newMockTask(name string) *mockTask {
 	return &mockTask{
 		pbTask: &pb.Task{
-			Name: name,
+			Name:  name,
+			State: pb.TaskState_PENDING,
+		},
+		finalState: pb.TaskState_SUCCESS,
+	}
+}
+
+func newMockFailingTask(name string) *mockTask {
+	return &mockTask{
+		pbTask: &pb.Task{
+			Name:  name,
+			State: pb.TaskState_PENDING,
 		},
+		finalState: pb.TaskState_FAILED,
+	}
+}
+
+func compareTaskStates(t *testing.T, tasks []Task, states []pb.TaskState) {
+	if len(tasks) != len(states) {
+		t.Errorf("`tasks` and `states` should have the same length (%d != %d)", len(tasks), len(states))
+	}
+
+	for i := range tasks {
+		task := tasks[i]
+		state := task.Proto(nil).State
+		if state != states[i] {
+			t.Errorf("Task `%s` expected to be in state %s, but was in %s instead.", task.Proto(nil).Name, states[i], state)
+		}
```

Comment:
A bit of Goism can be applied here:
```suggestion
	for i, task := range tasks {
		p := task.Proto(nil)
		
		if state := p.State; state != states[i] {
			t.Errorf("Task `%s` expected to be in state %s, but was in %s instead.", p.Name, states[i], state)
		}
```

I don't know about the assignment in the `if`, that's debatable (sometimes looks good, some other, and the use case varies), but for sure the `range` one is the way to Go (pun intended 😛 )